src/runtime/CollectionWrappers/SequenceWrapper.cs

```diff
@@ -0,0 +1,98 @@
+using System;
+using System.Collections.Generic;
+
+namespace Python.Runtime.CollectionWrappers
+{
+    internal class SequenceWrapper<T> : IterableWrapper<T>, ICollection<T>
+    {
+        public SequenceWrapper(PyObject pyObj) : base(pyObj)
+        {
+
+        }
+
+        public int Count
+        {
+            get
+            {
+                return (int)Runtime.PySequence_Size(pyObject.Handle);
+            }
+        }
+
+        public virtual bool IsReadOnly => false;
+
+        public virtual void Add(T item)
+        {
+            //not implemented for Python sequence.
+            //ICollection is the closest analogue but it doesn't map perfectly.
+            //SequenceWrapper is not final and can be subclassed if necessary
+            throw new NotImplementedException();
+        }
+
+        public void Clear()
+        {
+            if (IsReadOnly)
+                throw new NotImplementedException();
+            var result = Runtime.PySequence_DelSlice(pyObject.Handle, 0, Count);
+            if (result == -1)
+                throw new Exception("failed to clear sequence");
+        }
+
+        public bool Contains(T item)
+        {
+            //not sure if IEquatable is implemented and this will work!
+            foreach (var element in this)
+                if (element.Equals(item)) return true;
+
+            return false;
+        }
+
+        protected T getItem(int index)
+        {
+            IntPtr item = Runtime.PySequence_GetItem(pyObject.Handle, index);
+            object obj;
+
+            if (!Converter.ToManaged(item, typeof(T), out obj, true))
+            {
+                Runtime.XDecref(item);
+                Runtime.CheckExceptionOccurred();
+            }
+
+            return (T)obj;
+        }
+
+        public void CopyTo(T[] array, int arrayIndex)
+        {
+            for (int index = 0; index < Count; index++)
+            {
+                array[index + arrayIndex] = getItem(index);
```

Comment:
Not sure, but it might be more effective to use iterator.

Reply:
@lostmsu not disagreeing but please define effective.  Code clarity or runtime performance?

Reply:
@koubaa , I meant runtime performance. I am not sure if Python sequences must provide O(1) indexing.